dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,104 @@
+package com.datadog.android.rum.gestures
+
+import android.content.res.Resources
+import android.view.GestureDetector
+import android.view.MotionEvent
+import android.view.View
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.tracing.Tracer
+import java.lang.ref.WeakReference
+import java.util.concurrent.TimeUnit
+
+internal class DatadogGesturesListener(
+    private val rumTracer: Tracer,
+    private val decorViewReference: WeakReference<View>
+) :
+    GestureDetector.OnGestureListener {
+
+    // region GesturesListener
+
+    override fun onShowPress(e: MotionEvent) {
+        // No Op
+    }
+
+    override fun onSingleTapUp(e: MotionEvent): Boolean {
+        val decorView: View? = decorViewReference.get()
+        handleTapUpFor(decorView, e)
+        return false
+    }
+
+    override fun onDown(e: MotionEvent): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onFling(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        velocityX: Float,
+        velocityY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onScroll(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        distanceX: Float,
+        distanceY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onLongPress(e: MotionEvent) {
+        // No Op
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun handleTapUpFor(decorView: View?, e: MotionEvent) {
+        if (decorView != null) {
+            findTarget(decorView, e.x, e.y)?.let { target ->
+                // we just intercept but not steal the event
+                val spanBuilder = rumTracer
+                    .buildSpan(UI_TAP_ACTION_EVENT)
+                spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+                // we will handle general exceptions here. We don't know what might go wrong apart from
```

Comment:
This comment and the @Suppress are not needed anymore. And you don't handle the case where `getResourceEntryName()` returns null

Reply:
I am not sure the getResourceEntryName should return null but, I guess it will throw NotFoundException if it cannot find something. Just to be sure I will add a check there.